pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -154,15 +211,47 @@ func (m *manager) Start() {
 	}, 0)
 }
 
+func getCurrentCertificateOrBootstrap(
+	store Store,
+	bootstrapCertificatePEM []byte,
+	bootstrapKeyPEM []byte) (*CertKeyData, bool, error) {
+
+	certKeyData, err := store.Current()
+	if err == nil {
+		return certKeyData, false, nil
+	}
+
+	if _, ok := err.(*NoCertKeyError); !ok {
+		return nil, false, err
+	}
+
+	if bootstrapCertificatePEM == nil || bootstrapKeyPEM == nil {
+		return nil, false, fmt.Errorf("no cert/key available and no bootstrap cert/key to fall back to")
+	}
+
+	cert, err := tls.X509KeyPair(bootstrapCertificatePEM, bootstrapKeyPEM)
+	if err != nil {
+		return nil, false, err
+	}
+	certs, err := x509.ParseCertificates(cert.Certificate[0])
+	if err != nil {
+		return nil, false, fmt.Errorf("unable to parse certificate data: %v", err)
+	}
+	cert.Leaf = certs[0]
+	return &CertKeyData{&cert, bootstrapCertificatePEM, bootstrapKeyPEM}, true, nil
+}
+
 // shouldRotate looks at how close the current certificate is to expiring and
 // decides if it is time to rotate or not.
 func (m *manager) shouldRotate() bool {
 	m.certAccessLock.RLock()
 	defer m.certAccessLock.RUnlock()
-	notAfter := m.cert.Leaf.NotAfter
-	total := notAfter.Sub(m.cert.Leaf.NotBefore)
+	notAfter := m.certKeyData.Certificate.Leaf.NotAfter
+	total := notAfter.Sub(m.certKeyData.Certificate.Leaf.NotBefore)
```

Comment:
notAfter is certificate data. It's the expiration of the certificate. We can use it as you suggest. A follow up PR ok?